lib/utilities/retrieveData.js

```diff
@@ -47,24 +46,27 @@ export default function retrieveData(locations, retrieveDataFn, response, log) {
                 if (!responseDestroyed) {
                     _destroyResponse();
                 }
-                readable.emit('close');
+                readable.destroy();
                 return _next(responseErr);
             });
             // readable stream successfully consumed
-            readable.on('end', () => {
+            readable.once('end', () => {
+                readable.unpipe(response);
```

Comment:
is this unpipe necessary?

Reply:
@rahulreddy had added it before I got involved and we did not have a chance to talk about it since I only did the diff after his departure. It looks reasonable to me : we read the whole answer, so unpiping the response looks fair. Are you thinking about any side effect ?

Reply:
This was rando thing we tried. If you look at the implementation, unpipe just removes the writable from the list and emits 'unpipe'.